File: camerax_android/lib/src/video/recording.dart
import 'package:camerax_android/src/jni.dart' as jni;
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

class MyRecording implements Recording {
  final jni.Recording jniValue;

  MyRecording.jni(this.jniValue);

  @override
  void mute(bool muted) {
    jniValue.mute(muted);
  }

  @override
  void pause() {
    jniValue.pause();
  }

  @override
  void resume() {
    jniValue.resume();
  }

  @override
  void stop() {
    jniValue.stop();
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/video/Recording.kt
package dev.hebei.camerax_android.legacy.video

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiRecording

class Recording(registrar: CameraXRegistrar) : PigeonApiRecording(registrar) {
    override fun isPersistene(pigeon_instance: androidx.camera.video.Recording): Boolean {
        return pigeon_instance.isPersistent
    }

    override fun mute(pigeon_instance: androidx.camera.video.Recording, muted: Boolean) {
        return pigeon_instance.mute(muted)
    }

    override fun pause(pigeon_instance: androidx.camera.video.Recording) {
        return pigeon_instance.pause()
    }

    override fun resume(pigeon_instance: androidx.camera.video.Recording) {
        return pigeon_instance.resume()
    }

    override fun stop(pigeon_instance: androidx.camera.video.Recording) {
        return pigeon_instance.stop()
    }

    override fun close(pigeon_instance: androidx.camera.video.Recording) {
        return pigeon_instance.close()
    }
}